pegen/pegen.h

```diff
@@ -49,11 +54,12 @@ void *keyword_token(Parser *p, const char *val);
 
 void *CONSTRUCTOR(Parser *p, ...);
 
-#define LINE(arg) ((expr_ty)(arg))->lineno
-#define COL(arg) ((expr_ty)(arg))->col_offset
-#define ENDLINE(arg) ((expr_ty)(arg))->end_lineno
-#define ENDCOL(arg) ((expr_ty)(arg))->end_col_offset
-#define EXTRA(head, tail) LINE(head), COL(head), ENDLINE(tail), ENDCOL(tail), p->arena
+#define EXTRA_EXPR(head, tail) EXTRA(head, expr_type, tail, expr_type)
+#define EXTRA(head, head_type_func, tail, tail_type_func) head_type_func(head, 1, 1), \
```

Comment:
It seems a little perverse that the fundamental building block has the shorter name while the convenience macros have a longer name.